apps/frontend/src/app/PageTeam/CharacterDisplay/Tabs/TabOptimize/Components/ExcludeLoadout.tsx

```diff
@@ -0,0 +1,56 @@
+import { useDatabase, useOptConfig } from '@genshin-optimizer/gi/db-ui'
+import { CheckBox, CheckBoxOutlineBlank } from '@mui/icons-material'
+import { Button } from '@mui/material'
+import { useContext, useEffect } from 'react'
+import { TeamCharacterContext } from '../../../../../Context/TeamCharacterContext'
+
+export default function ExcludeLoadout({
+  disabled = false,
+}: {
+  disabled?: boolean
+}) {
+  const database = useDatabase()
+  const {
+    teamCharId,
+    teamChar: { optConfigId },
+    team,
+  } = useContext(TeamCharacterContext)
+  const { useLoadoutExclusion } = useOptConfig(optConfigId)!
+
+  useEffect(() => {
+    const artifactIds = Array.from(
```

Comment:
Move logic here into the filtering code called on optimize, since you want to recalculate teammates artifacts per optimization.